api/napi/__test__/copiler.spec.ts

```diff
@@ -0,0 +1,194 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+import test from 'ava'
+
+import { ComponentCompiler, ComponentDefinition, ComponentInstance, Property, ValueType } from '../index'
+
+test('get/set include paths', (t) => {
+  let compiler = new ComponentCompiler;
+
+  t.is(compiler.includePaths.length, 0);
+
+  compiler.includePaths = ["path/one/", "path/two/", "path/three/"];
+
+  t.deepEqual(compiler.includePaths, ["path/one/", "path/two/", "path/three/"]);
+})
+
+test('get/set style', (t) => {
+  let compiler = new ComponentCompiler;
+
+  t.is(compiler.style, null);
+
+  compiler.style = "fluent";
+  t.is(compiler.style, "fluent");
+})
+
+test('get/set build from source', (t) => {
+  let compiler = new ComponentCompiler;
+  let definition = compiler.buildFromSource(`export component App {}`, "");
+  t.not(definition, null);
+  t.is(definition!.name, "App");
+})
+
+test('constructor error ComponentDefinition and ComponentInstance', (t) => {
+  const componentDefinitionError = t.throws(() => { new ComponentDefinition });
+  t.is(componentDefinitionError?.message, "ComponentDefinition can only be created by using ComponentCompiler.");
+
+  const componentInstanceError = t.throws(() => { new ComponentInstance });
+  t.is(componentInstanceError?.message, "ComponentInstance can only be created by using ComponentCompiler.");
+})
+
+test('properties ComponentDefinition', (t) => {
+  let compiler = new ComponentCompiler;
+  let definition = compiler.buildFromSource(`export struct Struct {}
+  export component App {
+    in-out property <bool> bool-property;
+    in-out property <brush> brush-property;
+    in-out property <color> color-property;
+    in-out property <float> float-property;
+    in-out property <image> image-property;
+    in-out property <int> int-property;
+    in-out property <[string]> model-property;
+    in-out property <string> string-property;
+    in-out property <Struct> struct-property;
+  }`, "");
+  t.not(definition, null);
+
+  let properties = definition!.properties;
+  t.is(properties.length, 9);
+
+  properties.sort((a, b) => {
+    const nameA = a.name.toUpperCase(); // ignore upper and lowercase
+    const nameB = b.name.toUpperCase(); // ignore upper and lowercase
+
+    if (nameA < nameB) {
+      return -1;
+    }
+
+    if (nameA > nameB) {
+      return 1;
+    }
+
+    return 0;
+  })
+
+  t.is(properties[0].name, "bool-property");
```

Comment:
Unrelated to this particular review (think of it as todo for the future if we agree on it), but I'm starting to wonder if perhaps we should return an object with keys for each property by name, instead of an array. That way users can still iterate over the properties, but property access is much more convenient. @ogoffart What do you think?

Or perhaps the _real_ idiomatic API would be to return an iterator (that's a thing after all in JS). After all, that's also what the Rust API returns.